apps/src/webapp/formStorage.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * CodeOrgApp: Webapp
+ *
+ * Copyright 2014 Code.org
+ *
+ */
+
+'use strict';
+
+/**
+ * Namespace for form storage.
+ */ 
+var FormStorage = module.exports;
+
+
+/**
+ * Creates a new record in the specified table.
+ * @param {string} record.tableName The name of the table to read from.
+ * @param {Object} record Object containing other properties to store
+ *     on the record.
+ * @param {Function} callback Function to call with the resulting record.
+ */
+FormStorage.createRecord = function(record, callback) {
+  var tableName = record.tableName;
+  if (!tableName) {
+    console.log('readRecords: missing required property "tableName"');
+    return;
+  }
+  FormStorage.fetchTableSecret(
+      tableName, 
+      putRecord.bind(this, record, callback));
+};
+
+var putRecord = function(record, callback, tableSecret) {
+  var req = new XMLHttpRequest();
+  req.onreadystatechange = handlePutRecord.bind(req, record, callback, tableSecret);
+  var url = '//' + getFormDataHost() + '/v2/forms/CspTable/' + tableSecret +
+      '/children/CspRecord';
+  delete record.tableName;
+  var postData = {record_data_s: JSON.stringify(record)};
+  req.open('POST', url);
+  req.setRequestHeader("Content-Type", "application/json; charset=UTF-8");
+  req.send(JSON.stringify(postData));
+};
+
+var handlePutRecord = function(record, callback, tableSecret) {
+  if (this.readyState !== 4) {
+    return;
+  }
+  if (this.status !== 201) {
+    console.log('unexpected http status ' + this.status);
```

Comment:
It seems like the callback should be called in the event of an async error so that the user's program doesn't hang indefinitely with no error. perhaps the callback needs to have a status code in addition to the record?